### PR TITLE
Migrate CircleCI unit test workflow to Github Actions

### PR DESCRIPTION
**Pull request by shovnik**
## Which problem is this PR solving?
- This addresses [issue](https://github.com/open-telemetry/opentelemetry-js/issues/1259#event-4038172785) by migrating the remaining opentelemetry-js workflow from CircleCI to GitHub Actions (backwards compatibility, canary, docs and lint workflows are already on GHA)
- This also standardizes the triggers for CI workflows on opentelemetry-js

## Short description of the changes
- Migrates node-unit-tests and browser-unit-tests contained in config.yml file to Github Actions
- Standardized setting of triggers for all CI jobs to be on pull_request and push to master branch. Removed the current overlapping pull_request/push triggers.

## Migration Plan
I suggest having CircleCI and GitHub Action jobs run in parallel for a few weeks. After the GitHub Action jobs are running fine for a week or so and then remove the CircleCI workflows from config.yml

Note: Related PR migrating the corresponding config.yml file in the opentelemetry-js-contrib repo.

cc - @alolita

### CODE REVIEW

File: .github/workflows/unit-test.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      - name: Install Root Dependencies
+        run: npm install --ignore-scripts
+      - name: Boostrap Dependencies
+        run: npx lerna bootstrap --no-ci
+        env:
+          NPM_CONFIG_UNSAFE_PERM: true
+      - name: Unit tests
+        run: npm run test
+      - name: Report Coverage
+        run: npm run codecov
```
Comment (AzfaarQureshi) [inline .github/workflows/unit-test.yml]: maybe add a new line at the end of the file for consistency

File: packages/opentelemetry-instrumentation-xml-http-request/src/xhr.ts [hunk 2]
```diff
@@ -68,9 +68,7 @@ export interface XMLHttpRequestInstrumentationConfig
 /**
  * This class represents a XMLHttpRequest plugin for auto instrumentation
  */
-export class XMLHttpRequestInstrumentation extends InstrumentationBase<
-  XMLHttpRequest
-> {
+export class XMLHttpRequestInstrumentation extends InstrumentationBase<XMLHttpRequest> {
```
Comment (AzfaarQureshi) [inline packages/opentelemetry-instrumentation-xml-http-request/src/xhr.ts]: unrelated to GHA change

File: packages/opentelemetry-web/src/utils.ts [hunk 3]
```diff
@@ -125,9 +125,7 @@ export function getResource(
   startTimeHR: api.HrTime,
   endTimeHR: api.HrTime,
   resources: PerformanceResourceTiming[],
-  ignoredResources: WeakSet<PerformanceResourceTiming> = new WeakSet<
-    PerformanceResourceTiming
-  >(),
+  ignoredResources: WeakSet<PerformanceResourceTiming> = new WeakSet<PerformanceResourceTiming>(),
```
Comment (AzfaarQureshi) [inline packages/opentelemetry-web/src/utils.ts]: unrelated to GHA change

**Review by AzfaarQureshi — APPROVED** [review]
LGTM, seems like some random file changes creeped into the commit

File: .github/workflows/unit-test.yml (hunk 1 above)
Comment (shovnik) [reply .github/workflows/unit-test.yml]: Done

**Comment by shovnik** [comment]
Remove unrelated files added by linting

**Comment by shovnik** [comment]
Remove unrelated files added by linting

**Review by alolita — APPROVED** [review]
lgtm

**Comment by shovnik** [comment]
Filed upstream

**Pull request closed without merging**